### DISCUSSION TITLE
(Another) Pytorch question

### DISCUSSION THREAD

**Discussion by thatcort** [python-poetry/poetry#9123] [Q&A]
I'm having trouble installing the specific version of Pytorch that we need. I've read the replies to similar questions suggesting adding a source, but have not managed to get it to work.

Here is the pip installation command I am trying to duplicate:

```pip install --find-links https://download.pytorch.org/whl/torch_stable.html torch==2.1.0+cu121```

I've tried the following:
```
poetry source add -p explicit pytorch https://download.pytorch.org/whl/torch_stable.html
poetry add --source pytorch torch@2.1.0+cu121
```
But this results in an error: `403 Client Error: Forbidden for url: https://download.pytorch.org/whl/torch_stable.html/cu121/torch-2.1.0%2Bcu121-cp310-cp310-linux_x86_64.whl`

I'm unfortunately vague on the finer points of all the Pytorch versions and the inner workings of Poetry. Any help is appreciated!

**Comment by dimbleby** [comment]
#6885

**Reply by thatcort** [reply]
Sorry, that ticket seems unrelated to Pytorch. It's still unresolved, though, so should I take away that it is currently not possible to install Pytorch like this?
